Dockerfile

```diff
@@ -51,56 +51,69 @@ ADD https://github.com/tartley/colorama/archive/v0.3.3.tar.gz \
+    && apt update -y \
+    && apt install -y \
+       unzip \
+       python-setuptools=44.0.0-2 \
+       vera\+\+=1.2.1-2\+b5 \
+       shellcheck=0.7.1-1 \
+       gcc=4:9.2.1-3.1 \
+       make=4.2.1-1.2 \
+       jq \
+       cppcheck=1.90-4 \
+       frama-c-base=20191204+calcium-0.1 \
+    && apt autoremove -y \
+    && rm -rf /var/lib/apt/lists/* \
     && mkdir /home/sonarqube \
-    #Install i-code
-    && unzip /tmp/i-CodeCNES-3.1.0-CLI-linux.gtk.x86_64.zip -d /tmp;chmod +x /tmp/icode/icode;mv /tmp/icode/* /usr/bin \
+    ## Install i-Code CNES
+    && unzip /tmp/icode-4.1.0.zip -d /tmp \
+    && chmod +x /tmp/icode/icode \
+    && mv /tmp/icode/* /usr/bin \
     && rm -r /tmp/icode \
-    && rm /tmp/i-CodeCNES-3.1.0-CLI-linux.gtk.x86_64.zip \
-    ## Install sonar scanner
-    && unzip /tmp/scanners/sonar-scanner-cli-3.0.3.778-linux.zip -d /opt/ \
-    && mv /opt/sonar-scanner-3.0.3.778-linux /opt/sonar-scanner \
+    && rm /tmp/icode-4.1.0.zip \
+    ## Install Sonar Scanner
+    && unzip /tmp/scanners/sonar-scanner-cli-4.2.0.1873-linux.zip -d /opt/ \
+    && mv /opt/sonar-scanner-4.2.0.1873-linux /opt/sonar-scanner \
     && rm -rf /tmp/scanners \
-    ## Python, Pylint & CNES pylint setup
+    ## Python, Pylint & CNES Pylint setup
     && mkdir /opt/python \
     && find /tmp/python -maxdepth 1 -name \*.tar.gz -exec tar -xvzf {} -C /opt/python \; \
     && ls /opt/python \
-    && cd /opt/python/colorama-0.3.3/ && python setup.py install \
-    && cd /opt/python/python-lazy-object-proxy-1.2.1/ && python setup.py install \
-    && cd /opt/python/six-1.9.0/ && python setup.py install \
-    && cd /opt/python/wrapt-1.10.5/ && python setup.py install \
-    && cd /opt/python/astroid-astroid-1.4.9/ && python setup.py install \
-    && cd /opt/python/pylint-pylint-1.5/ && python setup.py install \
+    && cd /opt/python/colorama-0.3.3/ \
+    && python2 setup.py install \
+    && cd /opt/python/python-lazy-object-proxy-1.2.1/ \
+    && python2 setup.py install \
+    && cd /opt/python/six-1.9.0/ \
+    && python2 setup.py install \
+    && cd /opt/python/wrapt-1.10.5/ \
+    && python2 setup.py install \
+    && cd /opt/python/astroid-astroid-1.4.9/ \
+    && python2 setup.py install \
+    && cd /opt/python/pylint-pylint-1.5/ \
+    && python2 setup.py install \
     && rm -rf /tmp/python \
     ## C and C++ tools installation
-    && cd /tmp && tar -xvzf expat-2.0.1.tar.gz \
+    && cd /tmp \
+    && tar -xvzf expat-2.0.1.tar.gz \
     && cd expat-2.0.1 \
-    && ./configure && make && make install \
+    && ./configure \
+    && make \
+    && make install \
     && cd .. \
     && rm -rf ./expat-2.0.1.tar.gz ./expat-2.0.1 \
     && tar -xzvf rats-2.4.tgz \
     && cd rats-2.4 \
-    && ./configure --with-expat-lib=/usr/local/lib && make && make install \
+    && ./configure --with-expat-lib=/usr/local/lib \
+    && make \
+    && make install \
```

Comment:
At this step all C/C++ tools have been built from sources, so are there any apt build packages that become useless and thus could be removed from the Docker Image ?

Reply:
No rats and expats are still to be install, so make and gcc are needed. Package are still used.